sapl/base/views.py

```diff
@@ -915,6 +918,1229 @@ def get(self, request, *args, **kwargs):
+                        mandato.data_inicio_mandato,
+                        mandato.data_fim_mandato,
+                        prox_mandato.data_inicio_mandato,
+                        prox_mandato.data_fim_mandato)
+                    if exists:
+                        intersecoes.append(
+                            (parlamentar, mandato, prox_mandato))
+    return intersecoes
+
+
+class ListarParlMandatosIntersecaoView(PermissionRequiredMixin, ListView):
+    model = get_user_model()
+    template_name = 'base/parlamentares_mandatos_intersecao.html'
+    context_object_name = 'parlamentares_mandatos_intersecao'
+    permission_required = ('base.list_appconfig',)
+    paginate_by = 10
+
+    def get_queryset(self):
+        return parlamentares_mandatos_intersecao()
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            ListarParlMandatosIntersecaoView, self).get_context_data(**kwargs)
+        paginator = context['paginator']
+        page_obj = context['page_obj']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        context[
+            'NO_ENTRIES_MSG'
+            ] = 'Nenhum encontrado.'
+        return context
+
+
+def materias_protocolo_inexistente():
+    materias = []
+    for materia in MateriaLegislativa.objects.order_by('-ano', 'numero'):
+        if materia.numero_protocolo:
+            exists = Protocolo.objects.filter(
+                ano=materia.ano, numero=materia.numero_protocolo).exists()
+            if not exists:
+                materias.append(
+                    (materia, materia.ano, materia.numero_protocolo))
+    return materias
+
+
+class ListarMatProtocoloInexistenteView(PermissionRequiredMixin, ListView):
+    model = get_user_model()
+    template_name = 'base/materias_protocolo_inexistente.html'
+    context_object_name = 'materias_protocolo_inexistente'
+    permission_required = ('base.list_appconfig',)
+    paginate_by = 10
+
+    def get_queryset(self):
+        return materias_protocolo_inexistente()
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            ListarMatProtocoloInexistenteView, self
+            ).get_context_data(**kwargs)
+        paginator = context['paginator']
+        page_obj = context['page_obj']
+        context['page_range'] = make_pagination(
+            page_obj.number, paginator.num_pages)
+        context[
+            'NO_ENTRIES_MSG'
+            ] = 'Nenhuma encontrada.'
+        return context
+
+
+def protocolos_materias():
+    lista_protocolos_materias = []
+    protocolos = {}
+    
+    for m in MateriaLegislativa.objects.order_by('-ano', 'numero_protocolo'):
+        key = "{}/{}".format(m.numero_protocolo, m.ano)
+        val = protocolos.get(key, list())
+        val.append(m)
+        protocolos[key] = val
+
+    for k, v in protocolos.items():
+        if 'None' not in k:
+            if Protocolo.objects.filter(numero=int(k.split('/')[0]),
```

Comment:
Esses `splits` aqui não estão legais. Melhor colocar as seguintes linhas antes do `if`:
```
numero_protocolo = m.numero_protocolo
ano=m.ano
```

Reply:
Refatorei o código e retirei os `splits`

```suggestion
    def protocolos_com_materias():
    protocolos = {}
    
    for m in MateriaLegislativa.objects.filter(numero_protocolo__isnull=False).order_by('-ano', 'numero_protocolo'):
        if Protocolo.objects.filter(numero=m.numero_protocolo, ano=m.ano).exists():
            key = "{}/{}".format(m.numero_protocolo, m.ano)
            val = protocolos.get(key, list())
            val.append(m)
            protocolos[key] = val
            
    return [(v[0], len(v)) for (k, v) in protocolos.items() if len(v) > 1]
```